components/eamxx/tests/coupled/dynamics_physics/homme_shoc_cld_spa_p3_rrtmgp_pg2/CMakeLists.txt

```diff
@@ -0,0 +1,123 @@
+)
+
+# Set AD configurable options
+set (ATM_TIME_STEP 1800)
+SetVarDependingOnTestSize(NUM_STEPS 2 4 48)  # 1h 2h 24h
+set (RUN_T0 2021-10-12-45000)
+
+# Determine num subcycles needed to keep shoc dt<=300s
+set (SHOC_MAX_DT 300)
+math (EXPR MAC_MIC_SUBCYCLES "(${ATM_TIME_STEP} + ${SHOC_MAX_DT} - 1) / ${SHOC_MAX_DT}")
+
+## Copy (and configure) yaml files needed by tests
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/input.yaml
+               ${CMAKE_CURRENT_BINARY_DIR}/input.yaml)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/homme_shoc_cld_spa_p3_rrtmgp_pg2_output.yaml
+               ${CMAKE_CURRENT_BINARY_DIR}/homme_shoc_cld_spa_p3_rrtmgp_pg2_output.yaml)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/homme_shoc_cld_spa_p3_rrtmgp_pg2_gll_output.yaml
+               ${CMAKE_CURRENT_BINARY_DIR}/homme_shoc_cld_spa_p3_rrtmgp_pg2_gll_output.yaml)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/homme_shoc_cld_spa_p3_rrtmgp_pg2_diagnostics.yaml
+               ${CMAKE_CURRENT_BINARY_DIR}/homme_shoc_cld_spa_p3_rrtmgp_pg2_diagnostics.yaml)
+
+# Set homme's test options, so that we can configure the namelist correctly
+# Discretization/algorithm settings
+set (HOMME_TEST_NE 2)
+set (HOMME_TEST_LIM 9)
+set (HOMME_TEST_REMAP_FACTOR 3)
+set (HOMME_TEST_TRACERS_FACTOR 6)
+set (HOMME_TEST_TIME_STEP 300)
+set (HOMME_THETA_FORM 1)
+set (HOMME_TTYPE 5)
+set (HOMME_SE_FTYPE 2)
+set (HOMME_TEST_CUBED_SPHERE_MAP 2)
+set (HOMME_TRANSPORT_ALG 12)
+
+# Hyperviscosity settings
+set (HOMME_TEST_HVSCALING 0)
+set (HOMME_TEST_HVS 1)
+set (HOMME_TEST_HVS_TOM 0)
+set (HOMME_TEST_HVS_Q 6)
+
+set (HOMME_TEST_NU 7e15)
+set (HOMME_TEST_NUDIV 1e15)
+set (HOMME_TEST_NUTOP 2.5e5)
+
+# Testcase settings
+set (HOMME_TEST_MOISTURE notdry)
+set (HOMME_THETA_HY_MODE true)
+
+# Vert coord settings
+set (HOMME_TEST_VCOORD_INT_FILE acme-72i.ascii)
+set (HOMME_TEST_VCOORD_MID_FILE acme-72m.ascii)
+
+# Configure the namelist into the test directory
+configure_file(${SCREAM_SRC_DIR}/dynamics/homme/tests/theta.nl
+               ${CMAKE_CURRENT_BINARY_DIR}/namelist.nl)
+
+# Ensure test input files are present in the data dir
+set (TEST_INPUT_FILES
+  scream/maps/map_ne4np4_to_ne2np4_mono.nc
+  scream/init/spa_file_unified_and_complete_ne4_20220428.nc
+  scream/init/screami_ne4np4L72_20220823.nc
+  cam/topo/USGS-gtopo30_ne4np4pg2_16x_converted.c20200527.nc
+)
+foreach (file IN ITEMS ${TEST_INPUT_FILES})
+  GetInputFile(${file})
+endforeach()
+
+
+## Finally compare all MPI rank output files against the single rank output as a baseline, using CPRNC
+## Only if running with 2+ ranks configurations
+# This test requires CPRNC
+if (TEST_RANK_END GREATER TEST_RANK_START)
+  include (BuildCprnc)
+  BuildCprnc()
+
+  set (BASE_TEST_NAME "homme_shoc_cld_spa_p3_rrtmgp_pg2")
+  math (EXPR CMP_RANK_START ${TEST_RANK_START}+1)
+  foreach (MPI_RANKS RANGE ${CMP_RANK_START} ${TEST_RANK_END})
+
+    set (SRC_FILE "${BASE_TEST_NAME}_output.INSTANT.nsteps_x${NUM_STEPS}.np${MPI_RANKS}.${RUN_T0}.nc")
```

Comment:
If we only had one output nc file, this would be easier, no?

Reply:
This is the convention in some of the other tests. I'll combine for them as well all in a standalone commit. Unless you object since unrelated to this PR.